core/src/test/java/org/apache/iceberg/view/ViewCatalogTests.java

```diff
@@ -0,0 +1,1283 @@
+        .isEqualTo(
+            ImmutableViewVersion.builder()
+                .timestampMillis(view.currentVersion().timestampMillis())
+                .versionId(1)
+                .schemaId(EXPECTED_SCHEMA_ID)
+                .putSummary("operation", "create")
+                .defaultNamespace(identifier.namespace())
+                .addRepresentations(
+                    ImmutableSQLViewRepresentation.builder()
+                        .sql("select * from ns.tbl")
+                        .dialect("spark")
+                        .build())
+                .addRepresentations(
+                    ImmutableSQLViewRepresentation.builder()
+                        .sql("select * from ns.tbl using X")
+                        .dialect("trino")
+                        .build())
+                .build());
+
+    assertThat(catalog().dropView(identifier)).isTrue();
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+  }
+
+  @Test
+  public void createViewErrorCases() {
+    TableIdentifier identifier = TableIdentifier.of("ns", "view");
+
+    if (requiresNamespaceCreate()) {
+      catalog().createNamespace(identifier.namespace());
+    }
+
+    assertThat(catalog().viewExists(identifier)).as("View should not exist").isFalse();
+
+    SQLViewRepresentation trino =
+        ImmutableSQLViewRepresentation.builder()
+            .sql("select * from ns.tbl")
+            .dialect("trino")
+            .build();
+
+    // query is required
+    assertThatThrownBy(() -> catalog().buildView(identifier).create())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Cannot create view without specifying a query");
+
+    // schema is required
+    assertThatThrownBy(
+            () -> catalog().buildView(identifier).withQuery(trino.dialect(), trino.sql()).create())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Cannot create view without specifying schema");
+
+    // default namespace is required
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(identifier)
+                    .withQuery(trino.dialect(), trino.sql())
+                    .withSchema(SCHEMA)
+                    .create())
+        .isInstanceOf(IllegalStateException.class)
+        .hasMessage("Cannot create view without specifying a default namespace");
+
+    // cannot define multiple SQLs for same dialect
+    assertThatThrownBy(
+            () ->
+                catalog()
+                    .buildView(identifier)
+                    .withSchema(SCHEMA)
+                    .withDefaultNamespace(identifier.namespace())
+                    .withQuery(trino.dialect(), trino.sql())
+                    .withQuery(trino.dialect(), trino.sql())
+                    .create())
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Cannot add multiple SQLs for dialect: trino");
```

Comment:
Minor: SQL isn't really a noun, so this error message sounds odd to me. What about `"Invalid view version: Cannot add multiple queries for dialect %s"`?

Reply:
makes sense, I've updated that